iceoryx_hoofs/include/iceoryx_hoofs/cxx/expected.hpp

```diff
@@ -446,20 +392,6 @@ class IOX_NO_DISCARD expected<ValueType, ErrorType>
     /// @endcode
     const ValueType* operator->() const noexcept;
 
-    /// @brief conversion operator to an error only expected which can be useful
-    ///         if you would like to return only the success of a function
-    /// @return converts an expected which can contain a value and an error to an
-    ///         expected which contains only an error
-    /// @code
-    ///     cxx::expected<int, int> someErrorProneFunction(){}
-    ///
-    ///     cxx::expected<int> isItSuccessful() {
-    ///         return someErrorProneFunction();
-    ///     }
-    /// @endcode
-    template <typename T>
-    operator expected<T>() noexcept;
```

Comment:
Since it returns always a copy. And if we return a copy a non const overload produces just code bloat since the logic is identical to the const version.

You can always get a copy independent of the constness of the object.